x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -1070,6 +1081,144 @@ class AgentPolicyService {
     };
   }
 
+  public async createAgentlessAgent(
+    esClient: ElasticsearchClient,
+    soClient: SavedObjectsClientContract,
+    agentlessAgentPolicy: AgentPolicy
+  ) {
+    const logger = appContextService.getLogger();
+    logger.debug(`Creating agentless agent ${agentlessAgentPolicy.id}`);
+
+    if (!appContextService.getCloud()?.isCloudEnabled) {
+      logger.error('Creating agentless agent not supported in non-cloud environments');
+      throw new AgentlessAgentCreateError('Agentless agent not supported');
+    }
+    if (!agentlessAgentPolicy.supports_agentless) {
+      logger.error('Agentless agent policy does not have agentless enabled');
+      throw new AgentlessAgentCreateError('Agentless agent policy does not have agentless enabled');
+    }
+
+    const agentlessConfig = appContextService.getConfig()?.agentless;
+    if (!agentlessConfig) {
+      logger.error('Missing agentless configuration');
+      throw new AgentlessAgentCreateError('missing agentless configuration');
+    }
+
+    const policyId = agentlessAgentPolicy.id;
+    const { fleetUrl, fleetToken } = await this.getFleetUrlAndTokenForAgentlessAgent(
+      esClient,
+      policyId,
+      soClient
+    );
+
+    logger.debug(`Creating agentless agent with fleetUrl ${fleetUrl} and fleetToken ${fleetToken}`);
+
+    logger.debug(`Creating agentless agent with TLS config with certificate: ${agentlessConfig.api.tls.certificate},
+       and key: ${agentlessConfig.api.tls.key}`);
+
+    const tlsConfig = new SslConfig(
+      sslSchema.validate({
+        enabled: true,
+        certificate: agentlessConfig.api.tls.certificate,
+        key: agentlessConfig.api.tls.key,
+        certificateAuthorities: agentlessConfig.api.tls.ca,
+      })
+    );
+
+    const requestConfig = {
+      url: `${agentlessConfig.api.url}/deployments`,
+      data: {
+        policy_id: policyId,
+        fleet_url: fleetUrl,
+        fleet_token: fleetToken,
+        stack_version: appContextService.getKibanaVersion(),
+      },
+      method: 'POST',
+      headers: {
+        'Content-type': 'application/json',
+      },
+      httpsAgent: new https.Agent({
+        rejectUnauthorized: tlsConfig.rejectUnauthorized,
+        cert: tlsConfig.certificate,
+        key: tlsConfig.key,
+        ca: tlsConfig.certificateAuthorities,
+      }),
+    };
+
+    logger.debug(
+      `Creating agentless agent with request config ${JSON.stringify({
+        ...requestConfig,
+        httpsAgent: {
+          ...requestConfig.httpsAgent,
+          options: {
+            ...requestConfig.httpsAgent.options,
+            cert: requestConfig.httpsAgent.options.cert ? 'REDACTED' : undefined,
+            key: requestConfig.httpsAgent.options.key ? 'REDACTED' : undefined,
+            ca: requestConfig.httpsAgent.options.ca ? 'REDACTED' : undefined,
+          },
+        },
+      })}`
+    );
+
+    const response = await axios<AgentlessApiResponse>(requestConfig).catch(
```

Comment:
This can be left as a future to-do but the front-end needs to support redirects as well.